.github/workflows/csqa.yml

```diff
@@ -105,3 +105,37 @@ jobs:
 
       - name: Run Static Analysis
         run: composer static-analysis
+
+  perf-guard:
+    name: "Basic Performance Guard"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: "8.1"
+          coverage: none
+
+      # Install dependencies and handle caching in one go.
+      # Dependencies need to be installed to make sure the PHPUnit classes are recognized.
+      # @link https://github.com/marketplace/actions/install-php-dependencies-with-composer
+      - name: Install Composer dependencies
+        uses: "ramsey/composer-install@v3"
+        with:
+          # Bust the cache at least once a month - output format: YYYY-MM.
+          custom-cache-suffix: $(date -u "+%Y-%m")
+
+      - name: Download performance test fixture
+        run: wget https://raw.githubusercontent.com/WordPress/WordPress/refs/heads/master/wp-includes/PHPMailer/PHPMailer.php
+
+      - name: Run performance report
+        id: performance_report
+        run: echo "PHPCS_PERF_SECS=$(./vendor/bin/phpcs --standard=VariableAnalysis --report=Performance ./PHPMailer.php | grep -Eo 'TOTAL RUN TIME[ ]+[0-9.]+'|awk '{ print $4 }')" >> $GITHUB_OUTPUT
+
+      - name: Compare performance to baseline
+        if: ${{ fromJSON( steps.performance_report.outputs.PHPCS_PERF_SECS ) > 0.6 }}
```

Comment:
The output is not jsonified, so `fromJSON()` has no added benefit and may actually break things.

Reply:
I think this is the recommended way to compare numbers? https://github.com/sirbrillig/phpcs-variable-analysis/pull/345#discussion_r1866780299